src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -86,106 +69,233 @@ const openKeysToItem = (menuItems, selectedPath) => keysToItem(menuItems, select
 }, {});
 
 const CollapsingNavigationMenu = ({ selectedPath = undefined, menuItems, onSelect }) => {
-  const [isNewSelectedPath, setIsNewSelectedPath] = useState(false);
-  const [previousSelectedPath, setPreviousSelectedPath] = useState(selectedPath);
   const [openKeys, setOpenKeys] = useState(openKeysToItem(menuItems[0], selectedPath));
+  const currentNodeId = useRef();
+  const cursor = useRef(0);
   const selectedItem = useRef();
+  const previousSelectedPath = useRef(selectedPath);
   const theme = useContext(ThemeContext);
+  const visibleNodes = [];
 
-  if (previousSelectedPath !== selectedPath) {
-    setIsNewSelectedPath(true);
-    setOpenKeys({ ...openKeys, ...openKeysToItem(menuItems[0], selectedPath) });
-    setPreviousSelectedPath(selectedPath);
-  }
+  /**
+   * Assigns focus to current node.
+   */
+  const focusCurrentNode = () => {
+    const currentNode = currentNodeId.current ? document.getElementById(currentNodeId.current) : null;
+    if (currentNode) {
+      currentNode.focus();
+    }
+  };
 
+  /**
+   * Scrolls the currently selected menu item into view on mount.
+   * Ensures that the cursor is synched with the currently selected item.
+   */
   useEffect(() => {
+    const nodeId = selectedItem.current?.getAttribute('id');
+    const idx = visibleNodes.findIndex((el) => el.id === nodeId);
```

Comment:
Are there any edge cases where this could end up being `undefined === undefined` and return an improper index?

Reply:
The `menuItemPropType` declares that there must be a name attribute in that prop which contains a string. Because the nodeId is generated from the name attribute, it is guaranteed to at least be an empty string.